### PR TITLE
Updated tender import.

### PR DESCRIPTION
**Pull request by alexander-griffen**
This command pulls down _broadly_ correct and relevant data. That being said, I think that the better way to do it to get data that's more reliable is to pull down all (or most) of the data from the APIs.

Because this would take an extremely long time - at least initially - it's impractical to have this process happen with this command. It could be in an external repo that produces a csv that the import-tenders command could then ingest into the database. 

[I've written more about this, here.](https://docs.google.com/document/d/1d02ivH1lyzrtahR8R1BxC120ASvuz8OfHfvrJUcW5Jk/edit?usp=sharing)

### CODE REVIEW

File: procurement/management/commands/import_tenders.py [hunk 1]
```diff
@@ -21,46 +25,83 @@
...
-        )
-        parser.add_argument(
-            "--groups_file", action="store_true", help="provide alternative groups file"
+            "--get_all", action="store_true", help="Get all tender data"
         )
+        parser.add_argument("--update", action="store_true", help="Get new tender data")
 
     def handle(self, *args, **options):
         self.options = options
+        if self.check_valid_call():
```
Comment (struan) [inline procurement/management/commands/import_tenders.py]: You should be able to do this with the build in stuff for arg parse - https://docs.python.org/3/library/argparse.html#mutual-exclusion

File: procurement/management/commands/import_tenders.py [hunk 2]
```diff
@@ -71,102 +112,207 @@ def get_groups(self):
...
+        classification_descriptions_in_groups = self.groups.keys()
+        df = df.query(
+            "tender_classification_description in @classification_descriptions_in_groups"
+        )
+        # Next, merge with councils data to drop all rows not from a local authority
+        df = df.merge(
+            self.council_lookup, left_on="buyer_name", right_on="la-name", how="inner"
+        )
+        df = df.query("tender_status == 'complete'").reset_index(drop=True)
+        # Reindex the df with all required ccolumns so that if they don't show up
```
Comment (struan) [inline procurement/management/commands/import_tenders.py]: ccolumns typo :)

File: procurement/management/commands/import_tenders.py [hunk 3]
```diff
@@ -21,46 +25,83 @@
...
+    return f"https://mysociety.github.io/{repo}/data/{package}/{version}/{file}"
+
+
+def get_dataset_df(repo: str, package: str, version: str, file: str):
+    """
+    Get a dataframe from a dataset from the mysociety.github.io website.
+    """
+    url = get_dataset_url(repo, package, version, file)
+    return pd.read_csv(url)
+
```
Comment (struan) [inline procurement/management/commands/import_tenders.py]: there's a `mysoc_dataset` package that has both these methods in it.

**Review by struan — APPROVED** [review]
Looks good and seems to work. Some tiny reuse comments but otherwise looks good.

Might be nice to add a `@unittest.skip("awaiting updates")` to the relevant test class just to cut down on test fails.

**Review by zarino — APPROVED** [review]
Worked fine for me! (Running from a fresh database – before I noticed there was a `--get_all` option that would have dropped all the existing tenders for me, d’oh!)

I noticed a couple of contracts have a "total cost" of "£nan", eg:

- http://localhost:8000/contract/ocds-b5fd17-30dab93f-50af-4b90-af6f-fc95cad1b3c4/
- http://localhost:8000/contract/ocds-b5fd17-cbcbcc7a-cb30-4785-b6ad-b24b8e845693/

That last one is also an example of a number of contracts that start in the future (5th Dec), which I thought we were hiding from view? Others include:

- http://localhost:8000/contract/ocds-b5fd17-c4f4aa47-bd0a-4ce3-853c-8ce5e2339eeb/
- http://localhost:8000/contract/ocds-b5fd17-d1b2a1f6-2316-4e5f-95db-0678ff87ce0f/

**Pull request merged**